packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -170,12 +179,16 @@ class ImageCache {
       // Images that fail to load don't contribute to cache size.
       final int imageSize = info?.image == null ? 0 : info.image.height * info.image.width * 4;
       final _CachedImage image = _CachedImage(result, imageSize);
-      // If the image is bigger than the maximum cache size, and the cache size
-      // is not zero, then increase the cache size to the size of the image plus
-      // some change.
-      if (maximumSizeBytes > 0 && imageSize > maximumSizeBytes) {
-        _maximumSizeBytes = imageSize + 1000;
+
+      if (_isImageTooLarge(imageSize)) {
+        final LargeImageHandler handler = _largeImageHandler ?? _bumpUpMaximumSizeLargeImageHandler;
+        handler(this, imageSize);
+        if (_isImageTooLarge(imageSize)) {
+          // Abort insertion of image, it doesn't fit.
+          return;
```

Comment:
I think this should throw - but I also wonder if we could just restructure the signature of `handler` so that we never get here.

IOW, if we can make the Handler responsible for returning the thing we cache, we could give people the opportunity to either bump up the cache size or to resize the image to something that can be more reasonably cached.

Reply:
This isn't an exception.  If you hit this, it is the user's choice to disallow the caching of the image.